kmk/modules/modholdandtap.py

```diff
@@ -0,0 +1,65 @@
+from kmk.keys import make_argumented_key
+from kmk.modules import Module
+
+
+class ModHoldAndTapValidator:
+    def __init__(self, kc, mod):
+        self.kc = kc
+        self.mod = mod
+
+
+class ModHoldAndTap(Module):
+    def __init__(self):
+        self._timeout_key = False
+        self._active = False
+        self._prev_key = None
+        make_argumented_key(
+            names=('MHAT',),
+            validator=ModHoldAndTapValidator,
+        )
+
+    def during_bootup(self, keyboard):
+        return
+
+    def before_matrix_scan(self, keyboard):
+        return
+
+    def process_key(self, keyboard, key, is_pressed, int_coord):
+        if isinstance(key.meta, ModHoldAndTapValidator):
+
+            if self._active and self._prev_key is not None and is_pressed:
+                # release previous key
+                self.release(keyboard, self._prev_key)
+            self._prev_key = key
+            if is_pressed:
+                keyboard.process_key(key.meta.mod, is_pressed)
+
+                self._active = True
+            keyboard.process_key(key.meta.kc, is_pressed)
```

Comment:
```suggestion
            if self._active and self._prev_key is not None and is_pressed:
                # release previous key
                self.release(keyboard, self._prev_key)
            self._prev_key = key
            if is_pressed:
                keyboard.process_key(key.meta.mod, is_pressed)

                self._active = True
            keyboard.process_key(key.meta.kc, is_pressed)
```

This should go into `on_pressed` and `on_released` key handlers.